hack/stabilization-changes.py

```diff
@@ -210,6 +212,30 @@ def public_errata_uri(version, cache=None, **kwargs):
     return errata_uri, public
 
 
+def get_concerns_about_updating_out(version, channel, target_channel_versions, cache=None):
+    release_major_minor = '.'.join(version.split('.', 2)[:2])
+    try:
+        phase, channel_version = channel.rsplit('-', 1)
+    except ValueError:
+        return  # 'fast' and similar version-agnostic channels do not need updating-out concerns
+    if phase == 'candidate':
+        return  # we need ungated candidate-4.y admission to bootstrap using candidate-4.y update recommendations for gating later phases.
+    channel_major_minor = '.'.join(version.split('.', 2)[:2])
+    if release_major_minor == channel_major_minor:
+        return  # we are concerned about getting from 4.(y-1) and earlier into 4.y, not about movement within 4.y.
+    cincinnati_uri, cincinnati_data = get_cincinnati_channel(cache=cache, channel='candidate-{}'.format(channel_major_minor))
+    paths = FIXME # find version in cincinnati_data and look for updates (possibly multi-hop) to 4.y
```

Comment:
I think as long as `version` has a path to `release_major_minor + 1` we're good. In multi-hop we would've stopped 4.98.91 from going into eus-4.10 because 4.99.50 never made it in because it lacked an upgrade path to 4.100.

Reply:
Rebased and taught about path walking with 65ca645 -> 936de75.  I started in on your single-hop proposal, but realized that even if 4.99.50-to-4.100 was unconditionally recommended when we decided to promote 4.99.50, it may no longer be recommended when we are considering 4.98.91.  And holding 4.98.91 out of a 4.10 channel in that case will make it more likely that folks on 4.98.(z < 91) get a happy path through, instead of heading to 4.98.91, on to 4.99.50, and then being stuck.

Testing with a partial revert of #2245:

```console
$ git revert fd6471d
$ hack/stabilization-changes.py
...
* Recommend waiting to promote 4.7.54 to eus-4.8; it was promoted the feeder stable by 0a4c588e81 (Merge pull request #2237 from openshift-ota-bot/promote-4.7.54-to-stable, 2022-07-20, 4 days, 0:43:33.877052) No unconditional paths from 4.7.54 to later major or minor releases in https://api.openshift.com/api/upgrades_info/v1/graph?channel=candidate-4.8&arch=amd64
...
```

Which looks good to me.